src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -78,11 +78,20 @@ internal enum ContinuationFlags
         ContinueOnCapturedTaskScheduler = 64,
     }
 
+    // Keep in sync with dataAsyncResumeInfo in the JIT
+    internal unsafe struct ResumeInfo
+    {
+        public delegate*<Continuation, ref byte, Continuation?> Resume;
+        // IP to use for diagnostics. Points into main code and maps to the
```

Comment:
```suggestion
        // IP to use for diagnostics. Points into the jitted suspension code,
        // has an ASYNC native->IL mapping to the IL AsyncHelpers.Await call
        // which caused the suspension, and matches DiagnosticNativeOffset for
        // the corresponding AsyncSuspensionPoint in the debug info.
```

Reply:
I changed the wording to be more in line with your suggestion, but I also expanded a bit to explain the situation for optimized codegen.